.github/workflows/code-analysis.yml

```diff
@@ -1,47 +1,16 @@
 name: Code Analysis
-on: [push, pull_request]
-
-jobs:
-  call-build-workflow:
-    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name == github.repository
-    uses: ./.github/workflows/build.yml
-    with:
-      key_container: sonar
-    secrets: inherit
 
+on:
+  workflow_call:
 
+jobs:
   sonarcloud:
     if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name == github.repository
     name: Sonar Cloud code analysis
-    needs: call-build-workflow
     runs-on: ubuntu-latest
     env:
       SHARED: /tmp/workdir/pki
-    strategy:
-      matrix: ${{ fromJSON(needs.call-build-workflow.outputs.matrix) }}
     steps:
-      - name: Clone repository
-        uses: actions/checkout@v2
-
-      - name: Retrieve runner image
-        uses: actions/cache@v3
-        with:
-          key: pki-sonar-runner-${{ matrix.os }}-${{ github.run_id }}
-          path: pki-sonar-runner.tar
-
-      - name: Load runner image
-        run: docker load --input pki-sonar-runner.tar
-
-      - name: Set up PKI container
-        run: |
-          tests/bin/runner-init.sh pki
-       
-      - name: Copy build in current folder
-        run: docker cp pki:/usr/share/java/pki ./build
-      
-      - name: Remove maven related file
-        run: rm -f pom.xml
-
```

Comment:
In my test if sonar do not find che *.class files or if it find the pom.xml does not work. Is it working in your repository? This code job is performed only for the push or pull_request from the same repository and this happen when you merge the PR.